ApplicationLibCode/ProjectDataModel/Summary/RimSummaryPlot.cpp

```diff
@@ -1956,6 +1957,47 @@ void RimSummaryPlot::setAsCrossPlot()
     m_isCrossPlot = true;
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimSummaryPlot::handleDroppedObjects( const std::vector<caf::PdmObjectHandle*>& objects )
+{
+    auto sumCases = RimProject::current()->allSummaryCases();
+
+    for ( auto obj : objects )
+    {
+        auto summaryAdr = dynamic_cast<RimSummaryAddress*>( obj );
+        if ( summaryAdr )
+        {
+            if ( summaryAdr->isEnsemble() )
+            {
+                // TODO: Add drop support for ensemble curves
+            }
+            else
+            {
+                for ( auto sumCase : sumCases )
+                {
+                    if ( sumCase->caseId() == summaryAdr->caseId() )
+                    {
+                        auto* newCurve = new RimSummaryCurve();
+
+                        newCurve->setSummaryCaseY( sumCase );
+                        newCurve->setSummaryAddressYAndApplyInterpolation( summaryAdr->address() );
+
+                        addCurveNoUpdate( newCurve );
+
+                        newCurve->loadDataAndUpdate( true );
+
+                        break;
+                    }
```

Comment:
Maybe add method to find case by case id? Having to use "break" in a loop is usually a refactoring opportunity.